### ISSUE TITLE
Question about command line arguments for the UI version of the app

### ISSUE THREAD

**Issue by Virgus** [githubrobbi/Ultra-Fast-File-Search#3]
Hello @githubrobbi and thanks for the improvements of Swiftsearch in Ultra-Fast-File-Search.

UFFS is helping me in many ways and I'm mainly using the GUI version.
I just tried to improve my search scripts by trying to use command line parameters but none of those available for the CUI version seem to be interpreted by the GUI exe.

Is there an option to enable command line parameters also in the GUI version or they're not implemented ?
Or maybe I'm missing a command issued to the CUI version that triggers the output represented in the GUI ?

In case this use is not implemented yet, will there ever be a GUI version with all (or even part) of the options available for the CUI ?

Thanks again for this invaluable work,
kind regards,
V.

**Comment by githubrobbi** [comment]
Hi Virgus,

currently there is no processing of CL arguments when calling up the GUI.

The CLI version allows you to do everything and more of what the GUI version provides. Output can be console or a file.

Do you need to visualize the output and thus the need for the GUI? You can process the CLI output within your scrips. That is what I do in my processing pipelines.

Maybe if I understand your needs better, we can come up with a solution which works for you.

Getting the CL arguments processed by the GUI version takes a bit more time than I have right now.

**Comment by githubrobbi** [comment]
Hi @Virgus, good news and an honest status update.

I rewrote **UFFS** from scratch in Rust, and that version is available now:

https://github.com/skyllc-ai/UltraFastFileSearch

On Windows, you can install it with:

```bash
winget install SkyLLC.UFFS
```

Because the Rust version is my own codebase, I can finally build the project in the direction I originally wanted.

The GUI is next on my list, but it is not built yet, so I do not want to oversell it. The design you described is very close to what I want too: launch the GUI with CLI flags, prefill the search, and run that query automatically.

In other words, the GUI should respect the CLI instead of becoming a separate, disconnected tool.

For now, the Rust CLI and daemon are ready, and they already cover the old CLI workflows plus quite a bit more.

Thanks again for the kind words, and for bringing this up back in 2023.

**Comment by Virgus** [comment]
Hello, thanks for the message ! I'm seeing it only now, just by chance...

> In other words, the GUI should respect the CLI instead of becoming a separate, disconnected tool.
Looking forward to see your improvements as your new creation will grow :-)

Cheers,
V.
